### PR TITLE
TTS Captions

### PR DESCRIPTION
**Pull request by GrahamSH-LLK**
### Changes

Adds text to speech captions
image

### Reason for changes

It's widely requested and is a good "accessibility" patch on our shoulder
### Tests

Yes

### CODE REVIEW

**Comment by retronbv** [comment]
you seem to have left your sanity check in there

**Comment by fractioneater** [comment]
Shouldn't this have the community tag?

**Comment by mxmou** [comment]
> Shouldn't this have the community tag?

Indeed - that's why schema validation is failing.

**Comment by Secret-chest** [comment]
Could the [Sprite] tag be bold or something?

**Comment by mxmou** [comment]
> Could the [Sprite] tag be bold or something?

Is it even necessary? Text spoken by different sprites doesn't sound different in vanilla Scratch.

**Comment by Secret-chest** [comment]
> > Could the [Sprite] tag be bold or something?
> 
> Is it even necessary? Text spoken by different sprites doesn't sound different in vanilla Scratch.

It would be to distinguish it. Maybe make all sprite tags the same size so text gets aligned nicely.

**Comment by BroJac5246** [comment]
> > Could the [Sprite] tag be bold or something?
> 
> Is it even necessary? Text spoken by different sprites doesn't sound different in vanilla Scratch.

Maybe showing it should be a setting.

**Comment by CST1229** [comment]
> Is it even necessary? Text spoken by different sprites doesn't sound different in vanilla Scratch.

What about e.g different characters with different voices?

**Comment by Hans5958** [comment]
Weak opinion: The style of the captions needs to be done better. Something like [Google Meets' captions would work](https://workspaceupdates.googleblog.com/2019/04/live-captions-in-meet.html) comes in my mind, but anything works really.

File: addons/tts-captions/addon.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+      "url": "userscript.js",
+      "matches": ["projects"],
+      "runAtComplete": false
+    }
+  ],
+  "dynamicEnable": false,
+  "dynamicDisable": false,
+  "versionAdded": "1.30.0",
+  "tags": ["projectPage"],
```
Comment (mybearworld) [inline addons/tts-captions/addon.json:25]: This should have the community tag, as mentioned in https://github.com/ScratchAddons/ScratchAddons/pull/5514#issuecomment-1367882657.

```suggestion
  "tags": ["community", "projectPage"],
```

File: addons/tts-captions/userscript.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import createLogView from "../debugger/log-view.js";
+export default async function ({ addon, console }) {
+  console.log("sanity check");
+  const vm = addon.tab.traps.vm;
+  const logView = new createLogView();
+  console.log(logView);
+  logView.placeholderElement.textContent = "no logs";
```
Comment (mybearworld) [inline addons/tts-captions/userscript.js:7]: This should probably use `msg`.

File: addons/tts-captions/style.css [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  font-style: italic;
+}
+
+.sa-debugger-log-end {
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 1px;
+  height: 1px;
+}
```
Comment (mybearworld) [inline addons/tts-captions/style.css:31]: This is never used.

```suggestion
```

File: addons/tts-captions/userscript.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import createLogView from "../debugger/log-view.js";
```
Comment (mybearworld) [inline addons/tts-captions/userscript.js:1]: Does this really need to use debugger? I think this is overcomplicating it.

**Review by mybearworld — CHANGES_REQUESTED** [review]
My first review on GitHub, so I've probably done everything wrong. Also, the addon has just stopped working for me locally, so I can't really check if changes break anything.

**Comment by GrahamSH-LLK** [comment]
This doesn't work for dynamic values like joins and variables

**Comment by WorldLanguages** [comment]
I'm going to create an issue for this suggestion
Here it is: #6016

**Pull request closed without merging**